packages/swing-store/test/test-stateSync.js

```diff
@@ -116,10 +116,21 @@ test('export and import data for state sync', async t => {
 
   const exporter = makeSwingStoreExporter(dbDir);
 
+  const compareElems = (a, b) => a[0].localeCompare(b[0]);
+
   const kvData = [];
   for await (const elem of exporter.getExportData()) {
     kvData.push(elem);
   }
+  kvData.sort(compareElems);
+
+  const feedKVData = [];
+  for (const elem of exportLog.entries()) {
+    feedKVData.push(elem);
+  }
+  feedKVData.sort(compareElems);
+
+  t.deepEqual(kvData, feedKVData);
```

Comment:
I think this won't work if there are intermediate/transient values for the same key (including deletion).

I would have gone for `feedKVData = new Map()`, replicated set and deletions in it, and then checked that every entry in `getExportData()` exists as expected in `feedKVData`, finishing by comparing the number of entries between the 2.

Reply:
Sorry I missed the `shadowStore.entries()` part of this.

I think this may be a case where you want to expose the `shadowStore` directly, and simply do something like:

```js
  const kvData = [];
  const feedKVData = exportLog.data();
  for await (const elem of exporter.getExportData()) {
    kvData.push(elem);
    t.equal(feedKVData.get(elem[0]), elem[1]);
  }
  t.equal(feedKVData.size, kvData.length);
```

Reply:
No, because the order of things may not be precisely the same since either feed can have changes that get coalesced/superseded.  The ends states need to be the same though, so we save them up and compare those.  It's academic anyway because the code is finished and works.

Reply:
The code in my suggestion handles different order of entries by looking up and comparing the values corresponding to each keys, but yes I agree it's academic at this point.

My original concern was regarding the coalescing/superseding, which is handled by that shadowStore.